packages/react-components/react-toolbar/src/stories/ToolbarOverflow.stories.tsx

```diff
@@ -0,0 +1,204 @@
+  id: string;
+}
+
+export const TestOverflowMenuItem: React.FC<TestOverflowMenuItemProps> = props => {
+  const { id, ...rest } = props;
+  const isVisible = useIsOverflowItemVisible(id);
+
+  if (isVisible) {
+    return null;
+  }
+
+  if (id.includes('underline')) {
+    return <MenuItem icon={<TextUnderline16Regular />}> Underline</MenuItem>;
+  }
+
+  if (id.includes('bold')) {
+    return <MenuItem icon={<TextBold16Regular />}> Bold</MenuItem>;
+  }
+
+  if (id.includes('italic')) {
+    return <MenuItem icon={<TextItalic16Regular />}> Italic</MenuItem>;
+  }
+
+  if (id.includes('radio')) {
+    return (
+      <MenuList>
+        <MenuItemRadio name="fruits" value="1">
+          Apple
+        </MenuItemRadio>
+        <MenuItemRadio name="fruits" value="2">
+          Pear
+        </MenuItemRadio>
+        <MenuItemRadio name="fruits" value="3">
+          Banana
+        </MenuItemRadio>
+        <MenuItemRadio name="fruits" value="4">
+          Orange
+        </MenuItemRadio>
+      </MenuList>
+    );
+  }
+
+  return <MenuItem {...rest}>Item {id}</MenuItem>;
+};
+
+export const TestOverflowMenuDivider: React.FC<{
+  id: string;
+}> = props => {
+  const isGroupVisible = useIsOverflowGroupVisible(props.id);
+
+  if (isGroupVisible === 'visible') {
+    return null;
+  }
+
+  return <MenuDivider />;
+};
+
+export const OverflowMenu: React.FC<{ itemIds: Array<Array<string>> }> = ({ itemIds }) => {
+  const { ref, isOverflowing } = useOverflowMenu<HTMLButtonElement>();
+
+  if (!isOverflowing) {
+    return null;
+  }
+
+  return (
+    <Menu>
+      <MenuTrigger>
+        <Button ref={ref} icon={<MoreHorizontal20Filled />} appearance="subtle" />
+      </MenuTrigger>
+
+      <MenuPopover>
+        <MenuList>
+          {itemIds.map((group, groupIndx) => {
+            const isLast = groupIndx === itemIds.length - 1;
+            return (
+              <React.Fragment key={group.join()}>
+                {group.map(i => (
+                  <TestOverflowMenuItem key={i} id={i} />
+                ))}{' '}
+                {!isLast && <TestOverflowMenuDivider id={`${groupIndx + 1}`} />}
+              </React.Fragment>
+            );
+          })}
+        </MenuList>
+      </MenuPopover>
+    </Menu>
+  );
+};
+
+export type TestDividerProps = {
+  groupId: string;
+};
+
+export const TestDivider = ({ groupId }: TestDividerProps) => {
+  const groupVisibleState = useIsOverflowGroupVisible(groupId);
+
+  if (groupVisibleState !== 'hidden') {
+    return <ToolbarDivider />;
+  }
+
+  return null;
+};
+
+export const OverflowItems = (props: Partial<ToolbarProps>) => (
+  <div
+    style={{
+      resize: 'horizontal',
+      overflow: 'hidden',
+    }}
+  >
+    <Overflow padding={90}>
+      <Toolbar {...props} size="small">
+        <OverflowItem id="underline-1" groupId="1">
+          <ToolbarButton appearance="subtle" icon={<TextUnderline16Regular />} />
+        </OverflowItem>
```

Comment:
This is just some food for thought: what about creating components like `OverflowToolbarButton` and `OverflowToolbarRadioGroup` in this example so that the final JSX looks more compact and might reflect better what large apps would do ?

Reply:
Or even further, import `ToolbarButton` as `ToolbarButtonBase` and create a new `ToolbarButton` that wraps the `OverflowItem` around it

Reply:
That is interesting, the problem I see (other than creating confusion with so many toolbar components) is that internally `OverflowToolbarButton` would have to know which props pass through to which internal component. We can discuss it further